frontend/src/components/experimentRuns/ReadsPerSampleGraph.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useLayoutEffect, useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+import { Bar, BarChart, Tooltip, XAxis, YAxis } from 'recharts'
+import { useAppDispatch } from '../../hooks'
+import { FMSId, FMSSample } from '../../models/fms_api_models'
+import { loadReadsPerSample } from '../../modules/experimentRunLanes/actions'
+import { LaneInfo, NumberOfReads } from '../../modules/experimentRunLanes/models'
+import { list } from '../../modules/samples/actions'
+import { useResizeObserver } from '../../utils/ref'
+import { notifyError } from '../../modules/notification/actions'
+
+
+interface ReadsPerSampleGraphProps {
+	lane: LaneInfo
+}
+
+function ReadsPerSampleGraph({ lane }: ReadsPerSampleGraphProps) {
+	const DEFAULT_GRAPH_WIDTH = 800
+	const MIN_BAR_WIDTH = 6 // Make bars wide enough that the user can click them
+	const MAX_BAR_WIDTH = 64 // Don't let bars get too fat or the graph looks silly
+
+	const navigate = useNavigate()
+	const dispatch = useAppDispatch()
+	const { ref: resizeRef, size: componentSize } = useResizeObserver(800, 0)
+	const [graphWidth, setGraphWidth] = useState<number>(DEFAULT_GRAPH_WIDTH)
+
+	useEffect(() => {
+		if (!lane.readsPerSample) {
+			dispatch(loadReadsPerSample(lane.runName, lane.laneNumber))
+		}
+	}, [lane, dispatch])
+
+	useLayoutEffect(() => {
```

Comment:
> useLayoutEffect can hurt performance. Prefer [useEffect](https://react.dev/reference/react/useEffect) when possible.

Is this the risk you're willing to take 🤔

Reply:
useLayoutEffect is run before rendering. We have to compute the graph size before it is rendered, so this is the kind of thing useLayoutEffect was designed for.